pkg/gitconfig/configs.go

```diff
@@ -0,0 +1,358 @@
+// so those with a more narrow scope then only have to override what they are interested in.
+type Configs struct {
+	Preset   *Config
+	system   *Config
+	global   *Config
+	local    *Config
+	worktree *Config
+	env      *Config
+	workdir  string
+
+	SystemConfig   string
+	GlobalConfig   string
+	LocalConfig    string
+	WorktreeConfig string
+	EnvPrefix      string
+	NoWrites       bool
+}
+
+func New() *Configs {
+	return &Configs{
+		system: &Config{
+			readonly: true,
+		},
+		global: &Config{
+			path: globalConfigFile(),
+		},
+		local:    &Config{},
+		worktree: &Config{},
+		env: &Config{
+			noWrites: true,
+		},
+
+		SystemConfig:   systemConfig,
+		GlobalConfig:   globalConfig,
+		LocalConfig:    localConfig,
+		WorktreeConfig: worktreeConfig,
+		EnvPrefix:      envPrefix,
+	}
+}
+
+// Reload will reload the config(s) from disk.
+func (cs *Configs) Reload() {
+	cs.LoadAll(cs.workdir)
+}
+
+// LoadAll tries to load all known config files. Missing or invalid files are
+// silently ignored. It never fails. The workdir is optional. If non-empty
+// this method will try to load a local config from this location.
+func (cs *Configs) LoadAll(workdir string) *Configs {
+	cs.workdir = workdir
+
+	debug.Log("Loading gitconfigs for %+v ...", cs)
+
+	// load the system config, if any
+	if os.Getenv(cs.EnvPrefix+"_NOSYSTEM") == "" {
+		c, err := LoadConfig(cs.SystemConfig)
+		if err != nil {
+			debug.Log("failed to load system config: %s", err)
+		} else {
+			debug.Log("loaded system config from %s", cs.SystemConfig)
+			cs.system = c
+			cs.system.readonly = true
+		}
+	}
+
+	// load the "global" (per user) config, if any
+	cs.loadGlobalConfigs()
+	cs.global.noWrites = cs.NoWrites
+
+	// load the local config, if any
+	if workdir != "" {
+		localConfigPath := filepath.Join(workdir, cs.LocalConfig)
+		c, err := LoadConfig(localConfigPath)
+		if err != nil {
+			debug.Log("failed to load local config from %s: %s", localConfigPath, err)
+			// set the path just in case we want to modify / write to it later
+			cs.local.path = localConfigPath
+		} else {
+			debug.Log("loaded local config from %s", localConfigPath)
+			cs.local = c
+		}
+	}
+	cs.local.noWrites = cs.NoWrites
+
+	// load the worktree config, if any
+	if workdir != "" {
+		worktreeConfigPath := filepath.Join(workdir, cs.WorktreeConfig)
+		c, err := LoadConfig(worktreeConfigPath)
+		if err != nil {
+			debug.Log("failed to load worktree config from %s: %s", worktreeConfigPath, err)
+			// set the path just in case we want to modify / write to it later
+			cs.worktree.path = worktreeConfigPath
+		} else {
+			debug.Log("loaded local config from %s", worktreeConfigPath)
+			cs.worktree = c
+		}
+	}
+	cs.worktree.noWrites = cs.NoWrites
+
+	// load any env vars
+	cs.env = LoadConfigFromEnv(cs.EnvPrefix)
+
+	return cs
+}
+
+func globalConfigFile() string {
+	// $XDG_CONFIG_HOME/git/config
+	return filepath.Join(appdir.UserConfig(), "config")
```

Comment:
`appdir.UserConfig()` is currently returning `$XDG_CONFIG_HOME/gopass/config` or `$HOME/.config/gopass/config` :
https://github.com/gopasspw/gopass/blob/5981a5a643b682244b100c15a9aa48fb17a04621/pkg/appdir/appdir_xdg.go#L11-L23
and
https://github.com/gopasspw/gopass/blob/5981a5a643b682244b100c15a9aa48fb17a04621/pkg/appdir/appdir.go#L12-L13

So this here is actually returning `$XDG_CONFIG_HOME/gopass/config/config`, right?
I don't feel like it's what we want.

Reply:
This seems correct to me. We used to have the old yaml config in `$XDG_CONFIG_HOME/gopass/config/config.yml` and I just wanted to drop the extension.